spec/dockly/docker_spec.rb

```diff
@@ -3,6 +3,76 @@
 describe Dockly::Docker do
   subject { described_class.new(:name => :test_docker) }
 
+  describe '#registry' do
+    context 'when there is no registry' do
+      it 'returns nil' do
+        expect(subject.registry).to eq(nil)
+      end
+    end
+
+    context 'when there is only an ecr registry' do
+      subject do
+        Dockly::Docker.new do
+          ecr { server_address 'accountid.dkr.ecr.region.amazonaws.com' }
```

Comment:
Can pull this string out into a `let`

Reply:
The `let` syntax actually doesn't work inside the DSL, though I'm unclear why

Reply:
Hrm, ok!